packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -1284,7 +1284,7 @@ abstract class RenderSliver extends RenderObject {
   /// render object is to override its [hitTestSelf] and [hitTestChildren]
   /// methods.
   bool hitTest(SliverHitTestResult result, { required double mainAxisPosition, required double crossAxisPosition }) {
-    if (mainAxisPosition >= 0.0 && mainAxisPosition < geometry!.hitTestExtent &&
```

Comment:
This does not feel right, the paintExtent should be accurate represnt what is available on the screen.
I think the issue may be sliver grid is calculating paint extent incorrectly. It finds the delta of leadingScrollOffset and trailingScrollOffset. There is no problem with trailingScrollOffset, but it assume the leadingScrollOffset is always the scroll offset of firstchild. but that may not be true because it can be place arbitrary on screen.

Do you have the repro for the hittest issue?

Reply:
I think this is ok actually. The test I added includes the `hitTest` repro.

In the test case, there are 2 containers in the grid, one on top of the other, each with a height of 100. So the `paintExtent` is 200 when they are laid out, that seems right.

Checking `geometry.hitTest` here does not consider _where_ they are actually painted, even though it is being compared to the mainAxisPosition of the gesture here. In the test case the `leadingScrollOffset` is 300, the `paintExtent`/`hitTest` would have to be 500 to pass `hitTest` as it is, and that wouldn't be correct.

`SliverGeometry.hitTest` cannot exceed the `paintExtent`, and defaults to be the `paintExtent`, so it is 200 as well. This does not correlate with the actual position of the children, but is the correct representation of what has been painted.

It works when we know the children are placed at the top of the grid, but this is not allowing arbitrary placement. So the `hitTest` and `paintExtent` are correct, the question is, "have I hit within the area of the `SliverGrid` where it _may_ have painted a child?" - since we don't know yet where the grid children are positioned. I think the `maxPaintExtent` is a good representation of the area where a child may be painted. Once it passes this check, the children are `hitTest`, and this is where it reconciles the positioning of the children. WDYT?

Reply:
I thought of a better way to explain - the size of something does not necessary indicate its position.

Reply:
I think the current way of calculating the paint extent is not correct. according to the doc
```
The amount of currently visible visual space that was taken by the sliver
to render the subset of the sliver that covers all or part of the
[SliverConstraints.remainingPaintExtent] in the current viewport.
```
I attached a picture to demonstrate my thought.
![Untitled drawing (2)](https://user-images.githubusercontent.com/47866232/92029976-3823ed80-ed1b-11ea-81c9-e7d48c93a6b2.jpg)

Reply:
Interesting! I did not know the space should be included in the paintExtent. Let me try this and see how it comes out. :)